src/Week1/HW1.hs

```diff
@@ -0,0 +1,75 @@
+{-# OPTIONS_GHC -Wall #-}
+module Week1.HW1 where
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the last digit from a number
+lastDigit :: Integer -> Integer
+lastDigit n = mod n 10
+
+-- Drop the last digit from a number
+dropLastDigit :: Integer -> Integer
+dropLastDigit n = div n 10
+
+-- Exercise 2 -----------------------------------------
+-- Write a function that breaks apart a number into a list of its digits in reverse order
+
+toRevDigits :: Integer -> [Integer]
+toRevDigits num = helper num []
+  where helper n acc = if n <= 0 then revList' acc
+                                 else helper (dropLastDigit $ n) ((lastDigit n) :acc)
+
+-- helper: a function that reverses lists
+revList :: [a] -> [a]
+revList l = helper l [] 
+  where helper x acc = case x of []      -> acc
+                                 (x':xs) -> helper xs (x':acc)
+
+-- or, more concisely
+revList' :: [a] -> [a]
+revList' l = foldl (\acc x -> x:acc) [] l
+
+-- I also found this version on stackOverflow. I was a bit confused at first as to why foldl can simply take 2 arguments here. Now I think it's because the compiler `knows` that if revList'' takes in a list, its foldl definition must also take in a list, so we can just let the l be implicit (?)
```

Comment:
in fact it's more general than that; if you have a function 
```haskell
f x = someFunctionCombination x
```
then you can replace it with just
```haskell
f = someFunctionCombination
```

for example, in your function `lastDigit n = mod n 10` if we had a way to put `n` at the end of the line, we could use this trick; luckily we can turn any binary function (those that take two arguments) as an infix operator via backticks, so this is equivalent to writing
```haskell
lastDigit n = n `mod` 10
-- or equivalently
lastDigit n = (`mod` 10) n
-- and then with our trick
lastDigit = `mod` 10
```

Reply:
Thanks so much for the comments and the explanation! That's pretty cool :D